src/Aspire.Hosting.Redis/RedisInsightConfigWriterHook.cs

```diff
@@ -0,0 +1,104 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Hosting.ApplicationModel;
+using Aspire.Hosting.Lifecycle;
+using Polly;
+using System.Text.Json;
+
+namespace Aspire.Hosting.Redis;
+
+internal sealed class RedisInsightConfigWriterHook(IHttpClientFactory httpClientFactory, ResourceNotificationService resourceNotificationService) : IDistributedApplicationLifecycleHook
```

Comment:
Can you update this to use the new Eventing API?

Reply:
Sure.